### PR TITLE
Beadandó fájlok

### PR DESCRIPTION
**Pull request by BSX8GS**
main.tf, outputs.tf, provider.tf és variables.tf

### CODE REVIEW

File: Kiss Dorina Hedvig - BSX8GS/beadando/main.tf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        source = google_compute_disk.default.id
+    }
+  }
+
+  network_interface {
+    network = "default"
+  }
+}
+
+metadata_startup_script = "echo done > ~/test.txt"
```
Comment (s1mark) [inline Kiss Dorina Hedvig - BSX8GS/beadando/main.tf:27]: this should be a resource property

File: Kiss Dorina Hedvig - BSX8GS/beadando/main.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    attached_disk {
+        source = google_compute_disk.default.id
+    }
+  }
+
+  network_interface {
+    network = "default"
+  }
+}
```
Comment (s1mark) [inline Kiss Dorina Hedvig - BSX8GS/beadando/main.tf:25]: you need to assign the service account to the vm

**Comment by s1mark** [comment]
Looks ok.
Try running terraform plan and apply

**Comment by BSX8GS** [comment]
I ran terraform apply and that is the output i got:

Terraform used the selected providers to generate the following execution plan. 
Resource actions are indicated with the following symbols:
  + create
 <= read (data resources)

Terraform will perform the following actions:

  # data.google_compute_instance.vm_data will be read during apply
  # (depends on a resource or a module with changes pending)
 <= data "google_compute_instance" "vm_data" {
      + advanced_machine_features    = (known after apply)
      + allow_stopping_for_update    = (known after apply)
      + attached_disk                = (known after apply)
      + boot_disk                    = (known after apply)
      + can_ip_forward               = (known after apply)
      + confidential_instance_config = (known after apply)
      + cpu_platform                 = (known after apply)
      + current_status               = (known after apply)
      + deletion_protection          = (known after apply)
      + description                  = (known after apply)
      + desired_status               = (known after apply)
      + enable_display               = (known after apply)
      + guest_accelerator            = (known after apply)
      + hostname                     = (known after apply)
      + id                           = (known after apply)
      + instance_id                  = (known after apply)
      + label_fingerprint            = (known after apply)
      + labels                       = (known after apply)
      + machine_type                 = (known after apply)
      + metadata                     = (known after apply)
      + metadata_fingerprint         = (known after apply)
      + metadata_startup_script      = (known after apply)
      + min_cpu_platform             = (known after apply)
      + name                         = "bsx8gs"
      + network_interface            = (known after apply)
      + reservation_affinity         = (known after apply)
      + resource_policies            = (known after apply)
      + scheduling                   = (known after apply)
      + scratch_disk                 = (known after apply)
      + service_account              = (known after apply)
      + shielded_instance_config     = (known after apply)
      + tags                         = (known after apply)
      + tags_fingerprint             = (known after apply)
    }

  # google_compute_attached_disk.default will be created
  + resource "google_compute_attached_disk" "default" {
      + device_name = (known after apply)
      + disk        = (known after apply)
      + id          = (known after apply)
      + instance    = (known after apply)
      + mode        = "READ_WRITE"
      + project     = (known after apply)
      + zone        = (known after apply)
    }

  # google_compute_disk.default will be created
  + resource "google_compute_disk" "default" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "bsx8gs"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # google_compute_instance.default will be created
  + resource "google_compute_instance" "default" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done > ~/test.txt"
      + min_cpu_platform        = (known after apply)
      + name                    = "bsx8gs"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = "debian-cloud/debian-11"
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = "k.dorina33@gmail.com"
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # google_service_account.default will be created
  + resource "google_service_account" "default" {
      + account_id   = "bsx8gs-beadando"
      + disabled     = false
      + display_name = "Service Account of BSX8GS"
      + email        = (known after apply)
      + id           = (known after apply)
      + member       = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

Plan: 4 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + neptunCode = "bsx8gs"
  + nev        = "Kiss Dorina Hedvig"
  + vmNev      = "bsx8gs"

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_service_account.default: Creating...
google_compute_disk.default: Creating...
google_compute_instance.default: Creating...
google_service_account.default: Creation complete after 2s [id=projects/ekke-cloud/serviceAccounts/bsx8gs-beadando@ekke-cloud.iam.gserviceaccount.com]
google_compute_disk.default: Creation complete after 2s [id=projects/ekke-cloud/zones/europe-central2-a/disks/bsx8gs]
google_compute_instance.default: Still creating... [10s elapsed]
╷
│ Error: Error waiting for instance to create: The user does not have access to 
service account 'k.dorina33@gmail.com'.  User: 'k.dorina33@gmail.com'.  Ask a project owner to grant you the iam.serviceAccountUser role on the service account 
│ 
│
│   with google_compute_instance.default,
│   on main.tf line 11, in resource "google_compute_instance" "default":        
│   11: resource "google_compute_instance" "default" {

File: Kiss Dorina Hedvig - BSX8GS/beadando/main.tf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+  }
+
+  network_interface {
+    network = "default"
+  }
+
+    service_account {
+    email  = "k.dorina33@gmail.com"
```
Comment (s1mark) [inline Kiss Dorina Hedvig - BSX8GS/beadando/main.tf:28]: this line causes the issue in your apply.
There is no service account named `k.dorina33@gmail.com`
You should refer to your created service account resource

**Comment by BSX8GS** [comment]
Now terraform apply went well. I destroyed right after applying:

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create
 <= read (data resources)

Terraform will perform the following actions:

  # data.google_compute_instance.vm_data will be read during apply
  # (depends on a resource or a module with changes pending)
 <= data "google_compute_instance" "vm_data" {
      + advanced_machine_features    = (known after apply)
      + allow_stopping_for_update    = (known after apply)
      + attached_disk                = (known after apply)
      + boot_disk                    = (known after apply)
      + can_ip_forward               = (known after apply)
      + confidential_instance_config = (known after apply)
      + cpu_platform                 = (known after apply)
      + current_status               = (known after apply)
      + deletion_protection          = (known after apply)
      + description                  = (known after apply)
      + desired_status               = (known after apply)
      + enable_display               = (known after apply)
      + guest_accelerator            = (known after apply)
      + hostname                     = (known after apply)
      + id                           = (known after apply)
      + instance_id                  = (known after apply)
      + label_fingerprint            = (known after apply)
      + labels                       = (known after apply)
      + machine_type                 = (known after apply)
      + metadata                     = (known after apply)
      + metadata_fingerprint         = (known after apply)
      + metadata_startup_script      = (known after apply)
      + min_cpu_platform             = (known after apply)
      + name                         = "bsx8gs"
      + network_interface            = (known after apply)
      + reservation_affinity         = (known after apply)
      + resource_policies            = (known after apply)
      + scheduling                   = (known after apply)
      + scratch_disk                 = (known after apply)
      + service_account              = (known after apply)
      + shielded_instance_config     = (known after apply)
      + tags                         = (known after apply)
      + tags_fingerprint             = (known after apply)
      + zone                         = "europe-central2-a"
    }

  # google_compute_disk.default will be created
  + resource "google_compute_disk" "default" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-cloud/debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "bsx8gs-attached-disk"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # google_compute_instance.default will be created
  + resource "google_compute_instance" "default" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done > ~/test.txt"
      + min_cpu_platform        = (known after apply)
      + name                    = "bsx8gs"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + attached_disk {
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)
        }

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = "debian-cloud/debian-11"
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # google_service_account.default will be created
  + resource "google_service_account" "default" {
      + account_id   = "bsx8gs-beadando"
      + disabled     = false
      + display_name = "Service Account of BSX8GS"
      + email        = (known after apply)
      + id           = (known after apply)
      + member       = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + neptunCode = "bsx8gs"
  + nev        = "Kiss Dorina Hedvig"
  + vmNev      = "bsx8gs"

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.
  Enter a value: yes

google_service_account.default: Creating...
google_compute_disk.default: Creating...
google_service_account.default: Creation complete after 2s [id=projects/ekke-cloud/serviceAccounts/bsx8gs-beadando@ekke-cloud.iam.gserviceaccount.com]
google_compute_disk.default: Still creating... [10s elapsed]
google_compute_disk.default: Creation complete after 12s [id=projects/ekke-cloud/zones/europe-central2-a/disks/bsx8gs-attached-disk]
google_compute_instance.default: Creating...
google_compute_instance.default: Still creating... [10s elapsed]
google_compute_instance.default: Creation complete after 14s [id=projects/ekke-cloud/zones/europe-central2-a/instances/bsx8gs]
data.google_compute_instance.vm_data: Reading...
data.google_compute_instance.vm_data: Read complete after 1s [id=projects/ekke-cloud/zones/europe-central2-a/instances/bsx8gs]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

**Comment by s1mark** [comment]
Looks fine

**Pull request merged**